CMakeLists.txt

```diff
@@ -2,85 +2,53 @@ cmake_minimum_required(VERSION 3.15)
 
 # Set the project name to your project name, my project isn't very descriptive
 project(myproject CXX)
-include(cmake/StandardProjectSettings.cmake)
-include(cmake/PreventInSourceBuilds.cmake)
 
-# Link this 'library' to set the c++ standard / compile-time options requested
-add_library(project_options INTERFACE)
+# Add cmakelib
+include(FetchContent)
+FetchContent_Declare(cmakelib URL https://github.com/aminya/cmakelib/archive/refs/heads/main.zip)
+FetchContent_MakeAvailable(cmakelib)
+include(${cmakelib_SOURCE_DIR}/Index.cmake)
```

Comment:
Would it be an option to use `CPM.cmake` module to hide  `FetchContent`?
It works like this:
```cmake
# ---- Add dependencies via CPM ----
# see https://github.com/cpm-cmake/CPM.cmake for more info

include(cmake/CPM.cmake)

# PackageProject.cmake will be used to make our target installable
CPMAddPackage("gh:TheLartians/PackageProject.cmake@1.7.0")
# ...
packageProject(
  NAME ${PROJECT_NAME}
  VERSION ${PROJECT_VERSION}
  NAMESPACE ${PROJECT_NAME}
  BINARY_DIR ${PROJECT_BINARY_DIR}
  INCLUDE_DIR ${PROJECT_SOURCE_DIR}/include
  INCLUDE_DESTINATION ${INCLUDE_INSTALL_DIR}
  COMPATIBILITY SameMajorVersion
  # XXX DEPENDENCIES "fmt 7.1.3"
)
````

Reply:
for an usage example see this patch:
https://github.com/ClausKlein/cpp_starter_project/blob/3337f68032fbd0138edf91cfdbd09f7d9de59ae1/CMakeLists.txt#L29

Reply:
I prefer that we use built-in libraries. All we want to do is to somehow download cmakelib. I will eventually add git tags, so the downloading will be more reliable.

I think CPM is useful for actual libraries.